--- config.h.in.old	2005-10-18 02:32:04.000000000 +0400
+++ config.h.in	2007-03-31 16:32:03.000000000 +0400
@@ -9,6 +9,12 @@
 /* Define if debugging is enabled. */
 #undef CONFIG_DEBUG
 
+/* System-wide library path. */
+#undef CONFIG_LIBDIR
+
+/* Local library path. */
+#undef CONFIG_LOCALLIBDIR
+
 /* Default installation prefix. */
 #undef CONFIG_PREFIX
 
--- configure.old	2006-01-19 01:03:01.000000000 +0300
+++ configure	2007-03-31 16:32:49.000000000 +0400
@@ -1354,6 +1354,16 @@
 _ACEOF
 
 
+cat >>confdefs.h <<_ACEOF
+#define CONFIG_LIBDIR "$libdir"
+_ACEOF
+
+
+cat >>confdefs.h <<_ACEOF
+#define CONFIG_LOCALLIBDIR "${libdir/usr/usr/local}"
+_ACEOF
+
+
 
 # Check whether --enable-profile or --disable-profile was given.
 if test "${enable_profile+set}" = set; then
@@ -6411,6 +6445,7 @@
 	Source code location:	${srcdir}
 	Prefix			${ac_prefix}
 	Data path		${ac_prefix}/share/hydrogen
+	Library path		${libdir}
 	LDFLAGS			${ac_libraries}
 "
 if test "x$ac_debug" = "xdebug"; then
--- configure.in.old	2006-01-19 01:02:37.000000000 +0300
+++ configure.in	2007-03-31 16:32:37.000000000 +0400
@@ -36,6 +36,8 @@
 fi
 AC_SUBST(ac_prefix)
 AC_DEFINE_UNQUOTED(CONFIG_PREFIX, ["$ac_prefix"], [Default installation prefix.])
+AC_DEFINE_UNQUOTED(CONFIG_LIBDIR, ["$libdir"], [System-wide library path.])
+AC_DEFINE_UNQUOTED(CONFIG_LOCALLIBDIR, ["${libdir/usr/usr/local}"], [Local library path.])
 
 
 AC_ARG_ENABLE(profile, [  --enable-profile     enable profile info (default=no)], [useProfile="yes"], [useProfile="no"])
@@ -490,6 +492,7 @@
 	Source code location:	${srcdir}
 	Prefix			${ac_prefix}
 	Data path		${ac_prefix}/share/hydrogen
+	Library path		${libdir}
 	LDFLAGS			${ac_libraries}
 "
 if test "x$ac_debug" = "xdebug"; then
--- plugins/wasp/Makefile.in.old	2007-03-31 16:36:59.000000000 +0400
+++ plugins/wasp/Makefile.in	2007-03-31 16:37:55.000000000 +0400
@@ -1,7 +1,7 @@
 INCLUDES	=	-I/usr/local/include/ -Iinclude
 CC		=	gcc
 CFLAGS		=	$(INCLUDES) -fPIC -DPIC -Wall -O2 -g -c
-prefix  = @ac_prefix@
+libdir		=	@libdir@
 
 all: wasp_booster.so wasp_noisifier.so wasp_xshaper.so
 
@@ -23,9 +23,8 @@
 
 install:
 	@echo "Installing WASP plugins"
-	mkdir -p $(DESTDIR)$(prefix)/lib/hydrogen/plugins/
-	@install -vD *.so $(DESTDIR)$(prefix)/lib/hydrogen/plugins
+	mkdir -p $(DESTDIR)$(libdir)/hydrogen/plugins/
+	@install -vD *.so $(DESTDIR)$(libdir)/hydrogen/plugins
 
 uninstall:
-	@rm -rf $(DESTDIR)$(prefix)/lib/hydrogen
-	
+	@rm -rf $(DESTDIR)$(libdir)/hydrogen
--- src/lib/Preferences.cpp.old	2007-03-31 16:39:24.000000000 +0400
+++ src/lib/Preferences.cpp	2007-03-31 16:29:33.000000000 +0400
@@ -79,9 +79,9 @@
 		m_ladspaPathVect.push_back(sLadspaPath);
 	}
 	else {
-		m_ladspaPathVect.push_back( "/usr/lib/ladspa" );
-		m_ladspaPathVect.push_back( string( CONFIG_PREFIX ).append( "/lib/hydrogen/plugins" ) );
-		m_ladspaPathVect.push_back( "/usr/local/lib/ladspa" );
+		m_ladspaPathVect.push_back( CONFIG_LIBDIR "/ladspa" );
+		m_ladspaPathVect.push_back( CONFIG_LIBDIR "/hydrogen/plugins" );
+		m_ladspaPathVect.push_back( CONFIG_LOCALLIBDIR "/ladspa" );
 	}
 
 	m_pDefaultUIStyle = new UIStyle();
